massa-execution/src/worker.rs

```diff
@@ -255,106 +253,137 @@ impl ExecutionWorker {
     /// called when the blockclique changes
     fn blockclique_changed(
         &mut self,
-        mut blockclique: BlockHashMap<Block>,
-        mut finalized_blocks: BlockHashMap<Block>,
+        blockclique: BlockHashMap<Block>,
+        finalized_blocks: BlockHashMap<Block>,
     ) -> Result<(), ExecutionError> {
-        // filter out any already-finalized blocks
-        // this can happen for example on bootstrap because the consensus vs execution bootstrap snapshots may be out of sync
-        blockclique.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-        finalized_blocks.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-
         // stop the current VM execution and reset state to final
         // TODO make something more iterative/conservative in the future to reuse unaffected executions
         self.reset_to_final();
+        self.last_active_slot = self.last_final_slot;
```

Comment:
`/// Step 2: set last_active_slot = last_final_slot` (for example).

Reply:
done